analyzer.go

```diff
@@ -153,12 +156,48 @@ func (gosec *Analyzer) Process(buildTags []string, packagePaths ...string) error
 		Tests:      gosec.tests,
 	}
 
-	for _, pkgPath := range packagePaths {
-		pkgs, err := gosec.load(pkgPath, config)
-		if err != nil {
-			gosec.AppendError(pkgPath, err)
+	type result struct {
+		pkgPath string
+		pkgs    []*packages.Package
+		err     error
+	}
+
+	results := make(chan result)
+	jobs := make(chan string)
+
+	var wg sync.WaitGroup
+	wg.Add(gosec.concurrency)
+
+	worker := func(j chan string, r chan result) {
+		for s := range j {
+			packages, err := gosec.load(s, config)
+			r <- result{pkgPath: s, pkgs: packages, err: err}
+		}
+		wg.Done()
+	}
+
+	for i := 0; i < gosec.concurrency; i++ {
+		go worker(jobs, results)
+	}
+
+	go func() {
+		for _, pkgPath := range packagePaths {
+			jobs <- pkgPath
+		}
+
+		close(jobs)
+	}()
+
+	go func() {
```

Comment:
You can get rid of this goroutine if the Check is also concurrent and move the wg.Wait() at the end.